packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -459,13 +461,17 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     if (__DEV__) {
       if (fn.prototype && typeof fn.prototype.render === 'function') {
         const componentName = getComponentName(workInProgress);
-        warning(
-          false,
-          "The <%s /> component appears to have a render method, but doesn't extend React.Component. " +
-            'This is likely to cause errors. Change %s to extend React.Component instead.',
-          componentName,
-          componentName,
-        );
+
+        if (componentName !== null && !didWarnAboutBadClass[componentName]) {
```

Comment:
If it's null we still want to show the message. Let's just say `getComponentName(workInProgress) || 'Unknown'`

Reply:
I added that to make Flow happy, but your solution is better. I should do the same default on the server side warning too, correct?

Reply:
Yep, thanks!